scr/main/java/bittercode/service/impl/BookServiceImpl.java

```diff
@@ -0,0 +1,193 @@
+                int bQty = rs.getInt(5);
+
+                Book book = new Book(bCode, bName, bAuthor, bPrice, bQty);
+                books.add(book);
+            }
+        } catch (SQLException e) {
+
+        }
+        return books;
+    }
+
+    @Override
+    public String deleteBookById(String bookId) throws StoreException {
+        String response = ResponseCode.FAILURE.name();
+        Connection con = DBUtil.getConnection();
+        try {
+            PreparedStatement ps = con.prepareStatement(deleteBookByIdQuery);
+            ps.setString(1, bookId);
+            int k = ps.executeUpdate();
+            if (k == 1) {
+                response = ResponseCode.SUCCESS.name();
+            }
+        } catch (Exception e) {
+            response += " : " + e.getMessage();
+            e.printStackTrace();
+        }
+        return response;
+    }
+
+    @Override
+    public String addBook(Book book) throws StoreException {
+        String responseCode = ResponseCode.FAILURE.name();
+        Connection con = DBUtil.getConnection();
+        try {
+            PreparedStatement ps = con.prepareStatement(addBookQuery);
+            ps.setString(1, book.getBarcode());
+            ps.setString(2, book.getName());
+            ps.setString(3, book.getAuthor());
+            ps.setDouble(4, book.getPrice());
+            ps.setInt(5, book.getQuantity());
+            int k = ps.executeUpdate();
+            if (k == 1) {
+                responseCode = ResponseCode.SUCCESS.name();
+            }
+        } catch (Exception e) {
+            responseCode += " : " + e.getMessage();
+            e.printStackTrace();
+        }
+        return responseCode;
+    }
+
+    @Override
+    public String updateBookQtyById(String bookId, int quantity) throws StoreException {
+        String responseCode = ResponseCode.FAILURE.name();
+        Connection con = DBUtil.getConnection();
+        try {
+            PreparedStatement ps = con.prepareStatement(updateBookQtyByIdQuery);
+            ps.setInt(1, quantity);
+            ps.setString(2, bookId);
+            ps.executeUpdate();
+            responseCode = ResponseCode.SUCCESS.name();
+        } catch (Exception e) {
+            responseCode += " : " + e.getMessage();
+            e.printStackTrace();
+        }
+        return responseCode;
+    }
+
+    @Override
+    public List<Book> getBooksByCommaSeperatedBookIds(String commaSeperatedBookIds) throws StoreException {
+        List<Book> books = new ArrayList<Book>();
+        Connection con = DBUtil.getConnection();
+        try {
+            String getBooksByCommaSeperatedBookIdsQuery = "SELECT * FROM " + BooksDBConstants.TABLE_BOOK
+                    + " WHERE " +
+                    BooksDBConstants.COLUMN_BARCODE + " IN ( " + commaSeperatedBookIds + " )";
+            PreparedStatement ps = con.prepareStatement(getBooksByCommaSeperatedBookIdsQuery);
+            ResultSet rs = ps.executeQuery();
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/1#issuecomment-2418765539>#56fd03</a><span> - 10/17/2024, 07:44 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The current implementation of getBooksByCommaSeperatedBookIds method might be vulnerable to SQL injection attacks. It's advisable to review and potentially modify the approach for constructing the SQL query.
</div>
<div id="fix">
<b>Fix:</b> Use PreparedStatement with parameterized queries instead of concatenating the commaSeperatedBookIds directly into the SQL query. This will ensure proper escaping of user input and prevent SQL injection attacks.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -151,7 +151,8 @@
              String getBooksByCommaSeperatedBookIdsQuery = "SELECT * FROM " + BooksDBConstants.TABLE_BOOK
                      + " WHERE " +
 -                    BooksDBConstants.COLUMN_BARCODE + " IN ( " + commaSeperatedBookIds + " )";
 +                    BooksDBConstants.COLUMN_BARCODE + " IN (" +
 +                    String.join(",", Collections.nCopies(commaSeperatedBookIds.split(",").length, "?")) + ")";
              PreparedStatement ps = con.prepareStatement(getBooksByCommaSeperatedBookIdsQuery);
 +            String[] bookIds = commaSeperatedBookIds.split(",");
 +            for (int i = 0; i < bookIds.length; i++) {
 +                ps.setString(i + 1, bookIds[i].trim());
 +            }
              ResultSet rs = ps.executeQuery();
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged